src/main/java/com/mkul/game/ox/boardcheck/BoardChecker.java

```diff
@@ -0,0 +1,55 @@
+package com.mkul.game.ox.boardcheck;
+
+import com.mkul.game.ox.BoardScore;
+import com.mkul.game.ox.Field;
+import com.mkul.game.ox.FieldValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Class used to check if a win occurred
+ *
+ * @author Maciej Kulikiewicz
+ */
+public class BoardChecker {
+
+    private final List<BoardCheck> borderCheckList = new ArrayList<>();
+    private final BoardScore boardScore;
+    private final int winingSize;
+
+    public BoardChecker(BoardScore boardScore, int winingSize) {
+        this.boardScore = boardScore;
+        this.winingSize = winingSize;
+        borderCheckList.add(new BoardCheckHorizontal());
+        borderCheckList.add(new BoardCheckVertical());
+        borderCheckList.add(new BoardCheckFirstDiagonal());
+        borderCheckList.add(new BoardCheckSecondDiagonal());
```

Comment:
Można pomyśleć na zmianą tego na `List.of` i przypisywać całą listę w konstruktorze, albo wydzieleniu tego do metody prywatnej odpalanej z konstruktora.